File: go/src/88.合并两个有序数组.go
/*
 * @lc app=leetcode.cn id=88 lang=golang
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
func merge(nums1 []int, m int, nums2 []int, n int)  {
	index := m + n - 1
	m--
	n--
	for m >= 0 && n >= 0 {
		if nums1[m] >= nums2[n] {
			nums1[index] = nums1[m]
			m--
		} else {
			nums1[index] = nums2[n]
			n--
		}

		index--
	}

	for m >= 0 {
		nums1[index] = nums1[m]
		m--
		index--
	}

	for n >= 0 {
		nums1[index] = nums2[n]
		n--
		index--
	}
}
// @lc code=end

File: rust/src/88.合并两个有序数组.rs
/*
 * @lc app=leetcode.cn id=88 lang=rust
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        let mut cur = (m + n - 1) as usize;
        let mut m = m - 1;
        let mut n = n - 1;
        while (m >= 0 && n >= 0) {
            if (nums1[m as usize] > nums2[n as usize]) {
                nums1[cur] = nums1[m as usize];
                m -= 1;
            } else {
                nums1[cur] = nums2[n as usize];
                n -= 1;
            }
            cur -= 1;
        }

        while(n >= 0) {
            nums1[cur] = nums2[n as usize];
            cur -= 1;
            n -= 1;
        }
    }
}
// @lc code=end

